cpp/include/rapidsmpf/buffer/packed_data.hpp

```diff
@@ -47,6 +69,15 @@ struct PackedData {
     PackedData& operator=(PackedData&&) = default;
     PackedData(PackedData const&) = delete;
     PackedData& operator=(PackedData&) = delete;
+
+    /**
+     * @brief Check if the packed data is empty.
+     *
+     * @return True if the packed data is empty, false otherwise.
+     */
+    [[nodiscard]] inline bool empty() const {
```

Comment:
I think we should only use `inline` to obey the One Definition Rule (ODR)

```suggestion
    [[nodiscard]] bool empty() const {
```